### ISSUE TITLE
[Feature Request] Enhance class inheritance

### ISSUE THREAD

**Issue by lesmo** [akveo/react-native-ui-kitten#776]
### Issue type

* [ ] bug report
* [x] feature request

### Issue description

While attempting to augment functionality, or modify some components' behavior, one is unable to simply extend core Component classes because, most of the time, properties and methods are ```private```. This reduces the ability to add new functionality or tweaks on a "per project" basis.

It'd be nice to be able to add new methods or alter existing methods for custom functionality. It should be as simple as replacing ```private``` with the ```protected``` modifier.

**Comment by artyorsh** [comment]
What do you need to extend?

I don’t want to do this because:
1. Extending class components is not a good practice in React, it’s always better to achieve this result result with props
2. Sometime we will rewrite it with functional components 🤞

**Comment by lesmo** [comment]
Thanks for your reply!

Specifically, I was trying to make a `TopNavigation` component render any `ReactElement` in the title prop, and not just strings. The edit to the code to achieve this was quite simple, really... but it would be cool to be able to update ui-kitten and have this derived component get most of the new functionality or bug fixes with minimal code.

I read in another issue (that I couldn’t find again) that such is a little against the philosophy of the framework, so I just decided to hack it for my needs. Current composition API doesn’t allow for this particular case.

Currently I just copied the code to a new file in my project as a new component.
